src/lib.rs

```diff
@@ -221,3 +214,80 @@ pub unsafe extern "C" fn WrapperDecompressImageEx(
         }
     }
 }
+
+static GIT_VERSION: &str =
+    git_version::git_version!(args = ["--abbrev=40", "--always", "--dirty=-modified"]);
+
+static PACKAGE_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[no_mangle]
+pub unsafe extern "C" fn get_version(
+    package: &mut *const std::os::raw::c_char,
+    git: &mut *const std::os::raw::c_char,
+) {
+    *git = GIT_VERSION.as_ptr() as *const std::os::raw::c_char;
+    *package = PACKAGE_VERSION.as_ptr() as *const std::os::raw::c_char;
+}
+
+const DECOMPRESS_USE_16BIT_DC_ESTIMATE: u32 = 1;
+
+#[no_mangle]
+pub unsafe extern "C" fn create_decompression_context(features: u32) -> *mut std::ffi::c_void {
+    let enabled_features = if features & DECOMPRESS_USE_16BIT_DC_ESTIMATE != 0 {
+        EnabledFeatures::compat_lepton_vector_read()
+    } else {
+        EnabledFeatures::compat_lepton_scalar_read()
+    };
+
+    let context = Box::new(LeptonFileReader::new(enabled_features));
+    Box::into_raw(context) as *mut std::ffi::c_void
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn free_decompression_context(context: *mut std::ffi::c_void) {
+    let _ = Box::from_raw(context as *mut LeptonFileReader);
+    // let Box destroy the object
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn decompress_image(
+    context: *mut std::ffi::c_void,
+    input_buffer: *const u8,
+    input_buffer_size: u64,
+    input_complete: bool,
+    output_buffer: *mut u8,
+    output_buffer_size: u64,
+    result_size: *mut u64,
+) -> i32 {
+    match catch_unwind(|| {
+        let context = context as *mut LeptonFileReader;
+        let context = &mut *context;
```

Comment:
Could you please explain this Rust assignment, why it is needed?

Reply:
This turns the void* that is passed in into a rust mutable reference. I did it in two steps for clarity.